core/src/colliding_entities.rs

```diff
@@ -0,0 +1,129 @@
+use bevy::{prelude::*, utils::HashSet};
+
+use crate::CollisionEvent;
+
+/// Component which will be filled (if present) with a list of entities with which the current entity is currently in contact.
+#[derive(Component, Default, Reflect)]
+pub struct CollidingEntities(pub HashSet<Entity>);
```

Comment:
Keep the internal representation private.

Ofc that means we need to provide some useful public method. I would start with:

* `collided(&self) -> bool`
* `collided_with(&self, entity: Entity) -> bool`
* `collided_entities(&self) -> impl Iterator<Item = Entity>`

No need to go overboard, we may add more later when the need arise.

Reply:
Sounds reasonable. Will do it today after work.

Reply:
Done! But I decided to mirror the `HashSet` API to make it more familiar and consistent. What do you think?

Reply:
Thanks!

TBH, I am not 100% sure about mirroring the `HashSet` API. It may seem intuitive now because it is backed by a HashSet, but what about when it'll be backed by something else, maybe a `HashMap<Entity, CollisionData>`?

That being said, I don't want to bikeshed too much here either. This API is clear and effective, and if necessary, we'll be able to change it without being a breaking change (by adding new methods and deprecating the old ones)

So I only ask to change one thing:

* Make the iterator return owned values of `Entity` instead of reference (you can simply add a call of `.copied()` on the iterator). That's because `Entity` is a small `Copy` type and bevy API often (if not always) requires an owned value.

Reply:
Actually, the more I think about it, the more I think it's good like this. Good call 👍 

(Still change the iterator to return owned values ;-))

Reply:
But `HashSet` returns iterator with references. How can I convert it?

Reply:
> How can I convert it?

Yes: `self.0.iter().copied()`

Reply:
I think you forgot to address this point right?

Reply:
Oh, right, let me fix it.